### PR TITLE
Upgrade mkdocs to latest versions

### PR DESCRIPTION
**Pull request by rifelpet**
The biggest improvement I see is much snappier search results.

followed the material theme upgrade instructions [here](https://squidfunk.github.io/mkdocs-material/releases/5/)

### CODE REVIEW

**Comment by rifelpet** [comment]
this may need to wait until netlify updates the image tag used in their builds: https://github.com/netlify/build-image/releases/tag/v3.3.15 in order to support PYTHON_VERSION env var (its defaulting to 2.7 which doesn't have the newest modules we're upgrading to)

build log shows:

```
11:02:56 PM: build-image tag: v3.3.14
```

They push tags every couple weeks and the newest tag with PYTHON_VERSION support is only 6 days old, so hopefully it won't be long before builds start using it. alternatively we create a [runtime.txt](https://docs.netlify.com/configure-builds/manage-dependencies/#python) in the repo root but that's not very clean. I'd rather wait for the env var support.

/hold

**Comment by mikesplain** [comment]
Looks good, only holding an lgtm since we can’t see a preview, hopefully they’ll get that updated soon!

Thanks!

**Comment by rifelpet** [comment]
netlify updated their build environment to 3.3.15:

https://app.netlify.com/sites/kubernetes-kops/deploys/5ee9819edcd1790007bb6543

which supports specifying the python version by env var, as you can see it is now using python 3.7. The rest of the build succeeds and you can check out the deploy preview as normal.

I think this is ready for review/merge

File: images/mkdocs/Dockerfile [hunk 1]
```diff
@@ -14,14 +14,15 @@
 
 FROM alpine:3.10
 
-RUN apk update && apk add --no-cache \
+RUN apk add --no-cache \
```
Comment (hakman) [inline images/mkdocs/Dockerfile:17]: Why remove the `apk update` part?

Comment (rifelpet) [reply images/mkdocs/Dockerfile:17]: `add`'s `--no-cache` performs an `update` under the hood but doesnt cache the results. you can see the indexes being downloaded twice:

```
docker run -it --rm alpine
/ # apk update && apk add --no-cache bash
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
v3.12.0-74-gb8f92cf12a [http://dl-cdn.alpinelinux.org/alpine/v3.12/main]
v3.12.0-76-ge6ea27f814 [http://dl-cdn.alpinelinux.org/alpine/v3.12/community]
OK: 12734 distinct packages available
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.12/community/x86_64/APKINDEX.tar.gz
(1/4) Installing ncurses-terminfo-base (6.2_p20200523-r0)
(2/4) Installing ncurses-libs (6.2_p20200523-r0)
(3/4) Installing readline (8.0.4-r0)
(4/4) Installing bash (5.0.17-r0)
Executing bash-5.0.17-r0.post-install
Executing busybox-1.31.1-r16.trigger
OK: 8 MiB in 18 packages
```

also see [SO](https://stackoverflow.com/questions/49118579/alpine-dockerfile-advantages-of-no-cache-vs-rm-var-cache-apk)

Comment (hakman) [reply images/mkdocs/Dockerfile:17]: Thanks. Nice trick.

**Pull request merged**